src/Evolution/Systems/ScalarWave/Equations.hpp

```diff
@@ -130,25 +131,41 @@ struct UpwindFlux {
   // clang-tidy: non-const reference
   void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
 
+  // The fields for which the fluxes are computed. There must be a `Variables`
+  // with precisely these fields in the databox.
+  using target_fields = tmpl::list<Pi, Phi<Dim>, Psi>;
+
+  // This is the data needed to compute the numerical flux.
+  // `dg::SendBoundaryFluxes` calls `package_data` to store these tags in a
+  // Variables. Local and remote values of this data are then combined in the
+  // `()` operator.
   using package_tags =
       tmpl::list<Tags::NormalDotFlux<Pi>, Tags::NormalDotFlux<Phi<Dim>>, Pi,
                  NormalTimesFluxPi>;
 
-  using slice_tags = tmpl::list<Pi>;
+  // These tags are sliced to the interface of the element and passed to
+  // `package_data` to provide the data needed to compute the numerical fluxes.
+  using slice_tags =
+      tmpl::list<Tags::NormalDotFlux<Pi>, Tags::NormalDotFlux<Phi<Dim>>, Pi>;
```

Comment:
N.F is NOT sliced to the boundary because we don't have it in the volume. This makes absolutely no sense for hyperbolic systems.

Reply:
We should rename the list anyway (argument_tags like most other places?) since it isn't important that they are obtained by slicing.  In recent discussion about boundary conditions it has come up that we may want to use these functions, and it that case values will be directly computed on the boundary and not sliced.